tests/ci/cpu_unit_tests.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+variables:
+- group: AzureKeyVaultSecrets
+- name  : 'test'
+  value : 'tests/ci/run_pytest.py'
+- name  : 'testfolder'
+  value : './tests/unit'
+- name  : 'testmarkers'
+  value : '"not notebooks and not spark and not gpu"'
+- name  : 'junitxml'
+  value : 'reports/test-unit.xml'
+#   
+- name  : 'maxnodes'
+  value : 4
+#- name  : 'rg'
+#  value : 'recommender'
+#- name  : 'wsname'
+#  value : 'RecoWS'
+- name  : 'reponame'
+  value : 'Recommenders'
+- name  : 'branch'
+  value : 'azure-pipelines-bz'
+- name  : 'clustername'
+  value : 'Reco-cpu-test'
+# GPU
+#- name  : 'vmsize'
+#  value : 'STANDARD_NC6'
+# CPU
+- name  : 'vmsize'
+  value : 'STANDARD_D3_V2'
+- name  : 'dockerproc'
+  value : 'cpu'
+- name  : 'expname'
+  value : 'cpu_unit_tests'
+
+trigger: none
+
+pr:
+- staging 
+- master
+  
+pool:
+ vmImage: 'ubuntu-16.04'
+
+steps:
+
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '3.6'
+    architecture: 'x64'
+  displayName: 'Use Python 3.6'
+
+- script: | 
+   az login --service-principal -u $(ClientID) -p $(ClientSecret) --tenant $(TenantID)
+  displayName: 'Login to Azure'
+
+- bash: |
+    echo "##vso[task.prependpath]/data/anaconda/bin"
+  displayName: Add Conda to PATH
+
+- script: 
+     pip install azureml-sdk
+  displayName: 'install azureml-sdk'
+  
+- script: 
+     python scripts/generate_conda_file.py --name reco 
+  displayName: ' generate_conda_file.py'
+
+- script: |
+     chmod +x scripts/*.py
+     chmod +x tests/ci/*.py
+  displayName: 'ls'
+
+- script: |
+     pip install azure-cli
+     pip install --upgrade azureml-sdk[cli]

```

Comment:
probably we would save some time if we mix `pip install --upgrade azureml-sdk[cli]` with the previous `pip install azureml-sdk`

Reply:
In the next pr, common yml code is abstracted into a template.   I can test if time is saved and change it there.  As the template will be used in each pipeline, it's easier to make the change once.